packages/blade/src/components/Tag/Tag.stories.tsx

```diff
@@ -0,0 +1,159 @@
+    >
+      <Title>Usage</Title>
+      <Sandbox>
+        {`
+        import { Tag, FileTextIcon } from '@razorpay/blade/components';
+        
+        function App(): JSX.Element {
+          return (
+            <Tag
+              icon={FileTextIcon}
+              onDismiss={({ value }) => console.log('Tag dismissed', value)}
+            >
+              Unpaid
+            </Tag>
+          )
+        }
+
+        export default App;
+        `}
+      </Sandbox>
+    </StoryPageWrapper>
+  );
+};
+
+export default {
+  title: 'Components/Tag',
+  component: Tag,
+  argTypes: {
+    ...getStyledPropsArgTypes(),
+    icon: {
+      name: 'icon',
+      type: 'select',
+      options: Object.keys(iconMap),
+      mapping: iconMap,
+    },
+  },
+  parameters: {
+    docs: {
+      page: Page,
+    },
+  },
+} as Meta<TagProps>;
+
+const TagTemplate: ComponentStory<typeof Tag> = ({ children, ...args }) => {
+  return <Tag {...args}>{children}</Tag>;
+};
+
+export const Default = TagTemplate.bind({});
+Default.args = {
+  children: 'Unpaid',
+  onDismiss: ({ value }) => console.log('dismiss tag', value),
+  icon: 'FileTextIcon',
+} as TagProps & { icon: string };
+
+export const Disabled = TagTemplate.bind({});
+Disabled.args = {
+  children: 'Disabled Tag',
+  onDismiss: ({ value }) => console.log('dismiss tag', value),
+  icon: 'FileTextIcon',
+  isDisabled: true,
+} as TagProps & { icon: string };
+
+const CrossPlatformForm = ({
+  children,
+  onSubmit,
+}: {
+  children: React.ReactElement[];
+  onSubmit: DOMAttributes<HTMLFormElement>['onSubmit'];
+}): React.ReactElement => {
+  if (isReactNative()) {
+    return <Box>{children}</Box>;
+  }
+
+  return <form onSubmit={onSubmit}>{children}</form>;
+};
+
+export const ControlledTags = (props: TagProps): React.ReactElement => {
+  const [inputValue, setInputValue] = React.useState('');
+  const [tags, setTags] = React.useState<string[]>([]);
+
+  const addTag = (): void => {
+    // Add input value to tags and clear the input value
+    setTags([...tags, inputValue]);
+    setInputValue('');
+  };
+
+  const removeTag = (tagName: TagProps['children']): void => {
+    setTags(tags.filter((tagNameValue) => tagNameValue !== tagName));
+  };
+
+  return (
+    <Box>
+      <Box paddingY="spacing.4">
+        {tags.map((tagName) => (
+          <Tag
+            key={tagName}
+            {...props}
+            marginRight="spacing.2"
+            onDismiss={({ value }) => removeTag(value)}
+          >
+            {tagName}
+          </Tag>
+        ))}
+      </Box>
+      <Box>
+        <CrossPlatformForm
+          onSubmit={(e) => {
+            e.preventDefault();
+            addTag();
+          }}
+        >
+          <TextInput
+            label="Tag Label"
+            value={inputValue}
+            onChange={({ value }) => setInputValue(value ?? '')}
+            {...{ onSubmit: isReactNative() ? () => addTag() : undefined }}
```

Comment:
This is allowing me to add empty tags, which shouldn't even render since children prop is mandatory. Can we do an empty check here?